File: src/api/java/maestro/api/debug/IHudDebugRenderer.java
package maestro.api.debug;

import net.minecraft.client.gui.GuiGraphics;

/**
 * Interface for HUD debug rendering.
 *
 * <p>Handles 2D screen-space debug overlays.
 */
public interface IHudDebugRenderer {

    /**
     * Render HUD debug overlays.
     *
     * @param graphics the GUI graphics context for rendering
     * @param tickDelta partial tick time for smooth animations
     */
    void render(GuiGraphics graphics, float tickDelta);
}

File: common/src/main/maestro/api/debug/IHudDebugRenderer.kt
package maestro.api.debug

import net.minecraft.client.gui.GuiGraphics

/**
 * Interface for HUD debug rendering.
 *
 * Handles 2D screen-space debug overlays.
 */
interface IHudDebugRenderer {
    /**
     * Render HUD debug overlays.
     *
     * @param graphics the GUI graphics context for rendering
     * @param tickDelta partial tick time for smooth animations
     */
    fun render(
        graphics: GuiGraphics,
        tickDelta: Float,
    )
}
